trulens_eval/trulens_eval/streamlit.py

```diff
@@ -0,0 +1,274 @@
+
+tru = Tru()
+
+
+class FeedbackDisplay(BaseModel):
+    score: float = 0
+    calls: List[FeedbackCall]
+    icon: str
+
+
+def trulens_leaderboard(app_ids: List = None):
+    """
+    Render the leaderboard page.
+
+    Args:
+
+        app_ids : A list of application IDs (default is None)
+
+    !!! example
+
+        ```python
+        from trulens_eval import streamlit as trulens_st
+
+        trulens_st.trulens_leaderboard()
+        ```
+    """
+    tru = Tru()
+
+    lms = tru.db
+    df, feedback_col_names = lms.get_records_and_feedback([])
+    feedback_defs = lms.get_feedback_defs()
+    feedback_directions = {
+        (
+            row.feedback_json.get("supplied_name", "") or
+            row.feedback_json["implementation"]["name"]
+        ): row.feedback_json.get("higher_is_better", True)
+        for _, row in feedback_defs.iterrows()
+    }
+
+    if df.empty:
+        st.write("No records yet...")
+        return
+
+    df = df.sort_values(by="app_id")
+
+    if df.empty:
+        st.write("No records yet...")
+
+    apps = list(df.app_id.unique()) if app_ids is None else app_ids
+
+    for app in apps:
+        app_df = df.loc[df.app_id == app]
+        if app_df.empty:
+            continue
+        app_str = app_df["app_json"].iloc[0]
+        app_json = json.loads(app_str)
+        metadata = app_json.get("metadata")
+        st.header(app, help=draw_metadata(metadata))
+        app_feedback_col_names = [
+            col_name for col_name in feedback_col_names
+            if not app_df[col_name].isna().all()
+        ]
+        col1, col2, col3, col4, *feedback_cols = st.columns(
+            5 + len(app_feedback_col_names)
+        )
+        latency_mean = (
+            app_df["latency"].
+            apply(lambda td: td if td != MIGRATION_UNKNOWN_STR else None).mean()
+        )
+
+        col1.metric("Records", len(app_df))
+        col2.metric(
+            "Average Latency (Seconds)",
+            (
+                f"{millify(round(latency_mean, 5), precision=2)}"
+                if not math.isnan(latency_mean) else "nan"
+            ),
+        )
+        col3.metric(
+            "Total Cost (USD)",
+            f"${millify(round(sum(cost for cost in app_df.total_cost if cost is not None), 5), precision = 2)}",
+        )
+        col4.metric(
+            "Total Tokens",
+            millify(
+                sum(
+                    tokens for tokens in app_df.total_tokens
+                    if tokens is not None
+                ),
+                precision=2
+            ),
+        )
+
+        for i, col_name in enumerate(app_feedback_col_names):
+            mean = app_df[col_name].mean()
+
+            st.write(
+                styles.stmetricdelta_hidearrow,
+                unsafe_allow_html=True,
+            )
+
+            higher_is_better = feedback_directions.get(col_name, True)
+
+            if "distance" in col_name:
+                feedback_cols[i].metric(
```

Comment:
why don't you care about the color and stuff if it's distances?

Reply:
Distances are not bounded 0-1 and same thresholding doesn't apply. Also copied from `Leaderboard.py`